client/src/components/Upload.vue

```diff
@@ -16,16 +16,30 @@ function prepareFiles(files) {
   const csvFilter = (file) => /\.csv$/i.test(file.name);
   const imageFilter = (file) => imageFilesRegEx.test(file.name);
 
-  if (files.find(videoFilter)) {
-    return [
-      'video',
-      files.filter((file) => videoFilter(file) || csvFilter(file)),
-    ];
+  const videoFiles = files.filter(videoFilter);
+  const imageFiles = files.filter(imageFilter);
+  const csvFiles = files.filter(csvFilter);
+
+  if (videoFiles.length > 0 && imageFiles.length > 0) {
+    throw new Error('Do not upload images and videos in the same batch.');
+  } else if (csvFiles.length > 1) {
+    throw new Error('Can only upload a single CSV Annotation per import');
+  } else if (videoFiles.length > 1 && csvFiles.length > 0) {
+    throw new Error('Annotation upload is not supported when multiple videos are uploaded');
+  } else if (videoFiles.length) {
```

Comment:
Fails silently without user feedback if you try to upload a single CSV file.

Reply:
Fixed, ty